packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -101,6 +116,18 @@ export const useSignInContext = (): SignInContextType => {
   const { queryParams } = useRouter();
   const options = useOptions();
 
+  const initialValuesFromQueryParams = useMemo(() => {
+    const props: Record<string, string> = {};
+    const searchParams = new URLSearchParams(window.location.search);
+    searchParams.forEach((value, key) => {
+      if (SIGN_IN_INITIAL_VALUE_KEYS.includes(key) && typeof value === 'string') {
```

Comment:
❓ Do you need the type check at this point? Search params are always strings. Do you need to check if the value is != ''?